<launch>
    <arg name="online" default="true"/>
    <arg name="gui_rviz" default="false" />
    <arg name="gui_gazebo" default="true" />

    <rosparam file="$(find pomcp_ros)/params/shared_params.yml" command="load"/>

    <group if="$(eval arg('online') == true)">
        <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find pomcp_ros)worlds/rocksample.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="$(arg gui_gazebo)"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
        </include>

        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find pomcp_ros)/rviz/config.rviz" if="$(eval gui_rviz)"/>
    </group>


    <node name="world2odom" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 1 world odom 10"/>

    <node name="env" pkg="pomcp_ros" type="start_env" args="" respawn="false" output="screen">
        <param name="debug" type="bool" value="true" />
        <param name="rate" type="int" value="50" />

        <param name="seed" type="int" value="987654321" />
        <param name="half_efficiency_distance" type="double" value="10.0" />
        <param name="yaml_path" type="string" value="$(find pomcp_ros)/params/agent_start.yml" />

    </node>

</launch>